mppsolar/protocols/pi30m044.py

```diff
@@ -0,0 +1,473 @@
+            ],
+        ],
+        "test_responses": [
+            b"(2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 0 2 1>>\r",
+        ],
+    },
+    "QT": {
+        "name": "QT",
+        "description": "Device Time Inquiry",
+        "type": "QUERY",
+        "response_type": "SEQUENTIAL",
+        "response": [["bytes.decode:datetime.strptime(r, '%Y%m%d%H%M%S').strftime('%Y-%m-%d %H:%M:%S')", "Device Time", ""]],
+        "test_responses": [
+            b"(20210726122606JF\r",
+        ],
+    },
+    "QBMS": {
+        "name": "QBMS",
+        "description": "Read lithium battery information",
+        "help": " -- queries the value of various metrics from the battery",
+        "type": "QUERY",
+        "response_type": "SEQUENTIAL",
+        "crctype": "chk",
+        "response": [
+
+            [
+                "option",
+                "Battery connect status",
+                [
+                    "Connected",
+                    "Disconnected",
+                ],
+            ],
+            ["int", "Battery capacity from BMS", "%"],
+            [
+                "option",
+                "Battery force charging",
+                [
+                    "No",
+                    "Yes",
+                ],
+            ],
+            [
+                "option",
+                "Battery stop discharge flag",
+                [
+                    "Enable discharge",
+                    "Disable discharge",
+                ],
+            ],
+            [
+                "option",
+                "Battery stop charge flag",
+                [
+                    "Enable charge",
+                    "Disable charge",
+                ],
+            ],
+            ["float:r/10", "Battery bulk charging voltage from BMS", "V"],
+            ["float:r/10", "Battery float charging voltage from BMS", "V"],
+            ["float:r/10", "Battery cut-off voltage from BMS", "V"],
+            ["float", "Battery max charging current", "A"],
+            ["float", "Battery max discharge current", "A"]],
+         "test_responses": [
+            b"(0 100 0 0 1 532 532 450 0000 0030\x5E\n",
+        ],
+    },
+}
+
+SETTER_COMMANDS = {
+    "PBT": {
+        "name": "PBT",
+        "description": "Set Battery Type",
+        "help": " -- examples: PBT00 (set battery as AGM), Setting battery type, 00 for AGM, 01 for Flooded battery, 02 for user define, 03 for Pylontech, 04 for Shinheung, 05 for Weco, 06 for Soltaro, 07 for BAK, 08 for Lib, 09 for Lic",
+        "type": "SETTER",
+        "response": [["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}]],
+        "test_responses": [
+            b"(NAK\x73\x73\r",
+            b"(ACK\x39\x20\r",
+        ],
+        "regex": "PBT(0[0-9])$",
+    },
+}
+
+class pi30m044(pi30max):
+    def __str__(self):
+        return "PI30 protocol handler for PIP8048 and similar inverters"
```

Comment:
update to reflect target model, eg Axpert Max 7200 etc

Reply:
fixed